src/style-spec/function/evaluation_context.js

```diff
@@ -208,6 +235,15 @@ module.exports = () => ({
         }
 
         return interpolate[resultType](outputLower, outputUpper, t);
+    },
+
+    coalesce(args: Array<Function>) {
+        for (let i = 0; i < args.length - 1; i++) {
```

Comment:
Could be written as:

```js
let result = null;
for (const arg of args) {
    result = arg();
    if (result !== null) break;
}
return result;
```

Which handles the empty array case. (`coalesce` requires one argument currently, but we could consider relaxing this.)